src/vdom/index.js

```diff
@@ -11,18 +11,18 @@ export function isSameNodeType(node, vnode, hydrating) {
 		return node.splitText!==undefined;
 	}
 	if (typeof vnode.nodeName==='string') {
-		return !node._componentConstructor && isNamedNode(node, vnode.nodeName);
+		return !node._componentConstructor && isSameNodeName(node, vnode.nodeName);
 	}
-	return hydrating || node._componentConstructor===vnode.nodeName;
+	return hydrating || node._componentConstructor===vnode.nodeName || node.constructor === vnode.nodeName;
```

Comment:
This part won't fail, but https://github.com/developit/preact/pull/715/files/3d68b2af246be137c3be8b43f3ea1be84fe59af0#diff-4935bb00e75b7854383877cbfd9d770fR13 would since we new it there.

The worst that can happen at this point, using constructors, is allow it to get to the point of creation to fail because it wasn't registered.

One thing to note here, may be that `node.constructor` has the possibility of being `HTMLUnknownElement` if a tag name is used to construct it. This means that if `my-custom-element` goes unregistered, then `<my-custom-element />` is not the same thing as `<MyCustomElement />`, even though, theoretically they might share the same constructor, you just haven't associated them yet, thus causing it to try and construct the unregistered constructor.

Reply:
Something we're playing around in Skate with is autodefining custom elements that aren't defined. See: https://github.com/skatejs/skatejs/blob/master/packages/renderer-preact/src/index.js#L11. You most-probably don't want to do that here, but that's really the only thing we can do besides error with a helpful message that the custom element hasn't been defined yet.

Unfortunately the only good way to check that would be to get some movement on https://github.com/w3c/webcomponents/issues/566. Currently you have to pretty much try / catch, which I'm not sure you want to do in the critical path.

Reply:
whoo yeah that's hefty. I'll be keeping an eye on 566...